common/authentication/postgresql/metadata.go

```diff
@@ -133,6 +143,25 @@ func (m *PostgresAuthMetadata) GetPgxPoolConfig() (*pgxpool.Config, error) {
 			return nil
 		}
 	}
+	if m.UseAWSIAM {
+		awsRegion, ok := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSRegion")
+		if !ok {
+			return nil, fmt.Errorf("failed to find AWSRegion metadata field")
+		}
+		awsAccessKey, ok := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSAccessKey")
+		if !ok {
+			return nil, fmt.Errorf("failed to find AWSAccessKey metadata field")
+		}
+		awsSecretKey, ok := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSecretKey")
+		if !ok {
+			return nil, fmt.Errorf("failed to find AWSSecretKey metadata field")
+		}
```

Comment:
Here, you want to check not only that the metadata property exists, but also that it's not empty

```suggestion
	if m.UseAWSIAM {
		awsRegion, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSRegion")
		if awsRegion == "" {
			return nil, errors.New("metadata property AWSRegion is missing")
		}
		awsAccessKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSAccessKey")
		if awsAccessKey == "" {
			return nil, errors.New("metadata property AWSAccessKey is missing")
		}
		awsSecretKey, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSSecretKey")
		if awsSecretKey == "" {
			return nil, errors.New("metadata property AWSSecretKey is missing")
		}
```